Improve quality of bed name / number texture

yw4z opened the pull request:
This one is most noticeable while using printers with large print area (>320mm)

BEFORE - AFTER
![Screenshot-20250503215147](https://github.com/user-attachments/assets/93ba84cb-fb55-4b90-bbd0-5f9214611b51)

BEFORE - AFTER
![Screenshot-20250503214733](https://github.com/user-attachments/assets/c84c5b76-cd59-40fd-8f32-f5b53c412b39)
![Screenshot-20250503214757](https://github.com/user-attachments/assets/8e992c9d-ece4-4219-9fe2-a29f80d73ce9)

BEFORE - AFTER
![Screenshot-20250503214716](https://github.com/user-attachments/assets/30093d3a-7374-4898-8319-29527f858f19)
![Screenshot-20250503214659](https://github.com/user-attachments/assets/667127ac-ad32-4bb9-a9b9-da16878b2616)

Noisyfox commented on src/slic3r/GUI/PartPlate.cpp:
```diff
@@ -619,7 +619,7 @@ void PartPlate::calc_vertex_for_plate_name_edit_icon(GLTexture *texture, int ind
 
     if (texture && texture->get_width() > 0 && texture->get_height())
-        width = int(factor * (texture->get_original_width() * 16) / texture->get_height());
+        width = int(factor * (texture->get_original_width() * 10) / texture->get_height());
```
What does `10` mean here?

yw4z replied:
it changes its size. it rendered as much bigger while its value is 16 . i checked scaling on  windows.

Current build
![Screenshot-20250504124831](https://github.com/user-attachments/assets/4372515d-d94e-46b9-99ca-7c1c4415040c)

This PR
![Screenshot-20250504124823](https://github.com/user-attachments/assets/346cc6a7-f9a1-477b-9eb8-c03b3bd8330e)

I will recheck that ratio maybe we can use better ratio for that

yw4z commented:
@Noisyfox i fixes all problems with a bonus
problem was calculation of contour was wrong because of it changes number with next_highest_power_of_2. i changed this to ceil
```
    //m_width  = (int)next_highest_power_of_2((uint32_t)w);
    //m_height = (int)next_highest_power_of_2((uint32_t)h);
    m_width  = static_cast<int>(std::ceil(w)); 
    m_height = static_cast<int>(std::ceil(h));
```
texture map's frame ratio also was wrong you can see this example below. this image also at %100 scaling. you can see its like stretched

![Screenshot-20250505004322](https://github.com/user-attachments/assets/1d0857d2-138e-49b6-96dd-0a086ada3e55)

this image at %150 scaling. icon and text preserves their ratios
![Screenshot-20250505004322](https://github.com/user-attachments/assets/d4a2f2ad-b37c-4436-b48b-eca8d03bc6cf)

also updated edit icon files. text and edit icon centered vertically

matched format and simplified code

bonus part
centered plate name text on circular plates

![Screenshot-20250505005818](https://github.com/user-attachments/assets/800df40e-e3cd-42f6-80bc-671c06d8a04e)

this one was real reason to check code on name and number section

also i cant do that with previous contour calculation that gave weird spacing issue on long texts

![Screenshot-20250504223646](https://github.com/user-attachments/assets/80ddb1b4-73ab-4cd9-941d-f3638ed7983d)

here is its frame

![Screenshot-20250504223527](https://github.com/user-attachments/assets/333dc897-4ef8-443c-a5aa-64bd4a0685fc)

Noisyfox commented on src/slic3r/GUI/GLTexture.cpp:
```diff
@@ -492,8 +492,11 @@ bool GLTexture::generate_from_text(const std::string &text_str, wxFont &font, wx
+    //m_height = (int)next_highest_power_of_2((uint32_t)h);
+    // using next_highest_power_of_2 changes ratio when height / width values dont match
+    m_width  = static_cast<int>(std::ceil(w)); 
+    m_height = static_cast<int>(std::ceil(h));
```
This could cause problems on some GPUs, which requires texture sizes to be power of 2.

yw4z commented:
@Noisyfox can you check changes now. i got a simpler solution. but not sure about RAM usage if there are many plates i assume each one will generate new font. it can effect since sizes are very large

previous code uses offset_y wrong that causes streched text in axis. comparison in 100% scale
![Screenshot-20250506190813](https://github.com/user-attachments/assets/9a07ef69-2aa9-491e-a945-3a0ec16c04f5)

size stays same on %100, %150 and %200 scaling but its gives wrong size on %125, %175 and %225. i coulndt figure out why
![Screenshot-20250506221932](https://github.com/user-attachments/assets/4904e34d-266e-4b72-89c0-33168f480a63)

also reduced code changes and simplified calculations

dewi-ny-je commented:
@yw4z you a re doing an awesome job improving and making the UI more consistent. Thanks!

Noisyfox reviewed (approved):
LGTM, thanks

yw4z commented:
@Noisyfox did you read my comment about
"size stays same on %100, %150 and %200 scaling but its gives wrong size on %125, %175 and %225"

this required to fix that but this will broke render frame size calculation this time
```
    #if defined(__WXGTK__)
        // Linux specific issue : get_dpi_for_window(this) still doesn't responce to the Display's scale in new wxWidgets(3.1.3)
        scale_factor = wxGetApp().em_unit()
    #else
        scale_factor = 10.00 * ((double)get_dpi_for_window(wxGetApp().GetTopWindow()) / (double)DPI_DEFAULT);
    #endif // __WXGTK__
```

Noisyfox commented:
Yeah I did, but I think it works resasonably well according to my test, so I just merged it.

yw4z commented:
ok, i have tried completely different solutions but there are always problems while calculating text contour. i can share my final code if we got any more issues
